src/submodules/Lapack/src/GE_EigenValueMethods@Methods.F90

```diff
@@ -13,17 +13,181 @@
+!                                                                 getEig
+!----------------------------------------------------------------------------
+
+MODULE PROCEDURE zeig
+! LAPACK variables:
+INTEGER(I4B) :: info, lda, ldvl, ldvr, lwork, n, lrwork
+REAL(DFP), ALLOCATABLE :: rwork(:)
+COMPLEX(DFP), ALLOCATABLE :: vl(:, :), vr(:, :), work(:)
+
+CHARACTER(*), PARAMETER :: myName = "zeig"
+
+lda = SIZE(A(:, 1))
+n = SIZE(A(1, :))
+! CALL Assert(Mat=A, s=[n, n], msg="[ARG ERROR] :: A should be square", &
+!             file=__FILE__, line=__LINE__, routine=myName)
+! CALL Assert(n1=SIZE(lam), n2=n, msg="[ARG ERROR] :: size of lam should be "// &
+!             "equal to "//tostring(n), file=__FILE__, line=__LINE__, &
+!             routine=myName)
+! CALL Assert(mat=c, s=[n, n], msg="[ARG ERROR] :: shape of c should be"// &
+!             "the same as one of A", file=__FILE__, line=__LINE__, &
+!             routine=myName)
+ldvl = n
+ldvr = n
+lwork = 8 * n ! TODO: can this size be optimized? query first?
+lrwork = 2 * n
+ALLOCATE (vl(ldvl, n), vr(ldvr, n), work(lwork), rwork(lrwork))
+c = A
+CALL LA_GEEV('N', 'V', n, c, lda, lam, vl, ldvl, vr, ldvr, work, &
+             lwork, rwork, info)
+IF (info .NE. 0) CALL geevErrorMsg(info, n)
+c = vr
+
+END PROCEDURE zeig
+
+!----------------------------------------------------------------------------
+!                                                                 getEigVals
+!----------------------------------------------------------------------------
+
+MODULE PROCEDURE deigvals
+! LAPACK variables for DGEEV:
+REAL(DFP), ALLOCATABLE :: At(:, :), vl(:, :), vr(:, :), wi(:), work(:), wr(:)
+INTEGER :: info, lda, ldvl, ldvr, lwork, n
+
+CHARACTER(*), PARAMETER :: myName = "deigvals"
+
+lda = SIZE(A(:, 1))
+n = SIZE(A(1, :))
+! CALL assert_shape(A, [n, n], "solve", "A")
+ldvl = n
+ldvr = n
+lwork = 8 * n ! TODO: can this size be optimized? query first?
+ALLOCATE (At(lda, n), wr(n), wi(n), vl(ldvl, n), &
+          vr(ldvr, n), work(lwork))
+At = A
+
+CALL LA_GEEV('N', 'N', n, At, lda, wr, wi, vl, ldvl, vr, ldvr, &
+             work, lwork, info)
+IF (info .NE. 0) CALL geevErrorMsg(info, n)
+
+lam = wr + i_ * wi
+END PROCEDURE deigvals
+
+!----------------------------------------------------------------------------
+!                                                                getEigVals_2
+!----------------------------------------------------------------------------
+
+MODULE PROCEDURE zeigvals
+! LAPACK variables:
+INTEGER :: info, lda, ldvl, ldvr, lwork, n, lrwork
+REAL(DFP), ALLOCATABLE :: rwork(:)
+COMPLEX(DFP), ALLOCATABLE :: At(:, :), vl(:, :), vr(:, :), work(:)
+
+lda = SIZE(A(:, 1))
+n = SIZE(A(1, :))
+! CALL assert_shape(A, [n, n], "solve", "A")
+ldvl = n
+ldvr = n
+lwork = 8 * n ! TODO: can this size be optimized? query first?
+lrwork = 2 * n
+ALLOCATE (At(lda, n), vl(ldvl, n), vr(ldvr, n), &
+          work(lwork), rwork(lrwork))
+At = A
```

Comment:
Why do you make a copy

Reply:
This is because `GEEV` destroy the `A` 
I can add optional boolean argument like `destroy` to switch making copy.

For example 
```fortran 
INTERFACE GetEigVals
  MODULE SUBROUTINE deigvals(A, lam)
    REAL(DFP), INTENT(IN) :: A(:, :)
    COMPLEX(DFPC), INTENT(INOUT) :: lam(:)
    LOGICAL(LGT), OPTIONAL, INTENT(IN) :: destroy
    ! default true
  END SUBROUTINE deigvals
END INTERFACE GetEigVals
```
How do you think ?